pkg/process/monitor/process_monitor.go

```diff
@@ -184,14 +190,35 @@ func (pm *ProcessMonitor) initNetlinkProcessEventMonitor() error {
 func (pm *ProcessMonitor) initCallbackRunner() {
 	cpuNum := runtime.NumVCPU()
 	pm.callbackRunner = make(chan func(), pendingCallbacksQueueSize)
+	pm.callbackRunnerStopChannel = make(chan struct{})
 	pm.callbackRunnersWG.Add(cpuNum)
 	for i := 0; i < cpuNum; i++ {
+		// Copy i to avoid unexpected behaviors
+		callbackRunnerIndex := i
 		go func() {
 			defer pm.callbackRunnersWG.Done()
-			for call := range pm.callbackRunner {
-				if call != nil {
-					pm.tel.callbackExecuted.Add(1)
-					call()
+			for {
+				// We utilize the callbackRunnerStopChannel to signal the stopping point,
+				// as closing the callbackRunner channel could lead to a panic when attempting to write to it.
+
+				// Trying to exit the goroutine as early as possible.
+				// This is essential because of how the Go select statement functions. if both cases evaluate to true, it will randomly choose between the two.
```

Comment:
then, why do we need the duplication on the second select?
seems like having 2 different selects statements gives you the confidence you need

Reply:
When only the second select statement is present and we set the callbackRunnerStopChannel, there's a 50% chance that the second case will be chosen due to the workings of the select mechanism in Go. This is why I've introduced the first select statement—to attempt early termination of the goroutine (drawing inspiration from https://go101.org/article/channel-closing.html)

Reply:
I think you should add the source for the inspiration to the comment